src/collapse/dtrace.rs

```diff
@@ -234,38 +351,71 @@ impl Folder {
             }
         }
         // count it!
-        *self.occurrences.entry(stack_str).or_insert(0) += count;
+        self.occurrences.add(stack_str, count);
+
         // reset for the next event
         self.stack_str_size = 0;
         self.stack.clear();
     }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+#[cfg(test)]
+mod tests {
```

Comment:
This should probably also have a multi-threaded test like the `perf` version does.